packages/itwinui-css/src/table/base.scss

```diff
@@ -418,7 +418,7 @@
 
   &.iui-table-cell-sticky {
     position: sticky;
-    z-index: 1;
+    z-index: 2; // Above the non-sticky columns and the table resizer
```

Comment:
I just noticed that this doesn't work for resizers inside sticky columns.

Reply:
Screen recording shows how you can hover near left edge of the next non-sticky column to resize.

https://github.com/iTwin/iTwinUI/assets/9084735/02487d9c-7ee2-4761-a85e-1fca69141e52

Reply:
Actually, that issue is not originating from the changes in this PR. This issue can be seen in the currently hosted [v3 Full2 story](https://itwin.github.io/iTwinUI/react/?story=table--full2&theme=dark) and also in the [v2 Full2 story](https://itwin.github.io/iTwinUI/legacy/v2/react/?path=/story/core-table--full-2).

I will try fixing it in this PR if it's a small fix. But if it's a larger fix, it may be better to have a new PR for that. Investigating it further.

Reply:
Maybe we can try something like:
1. increase the resizer's z-index to be higher than sticky columns
2. decrease the column and resizer's z-indices when the column is being dragged (untested)

Not sure about 2 but I think 1 is definitely needed, because otherwise the resizer's hit target is like 8px (rather than 16px), which makes it difficult to use.

Reply:
Another suggestion: changing `:hover` selectors to `:hover:not(:active)` to avoid triggering hover styles when dragging.

Reply:
As discussed, I will work on this in a future PR. Added this to the After PR TODOs.